websites/A/Animes Portal/presence.ts

```diff
@@ -0,0 +1,294 @@
+const presence = new Presence({
+  clientId: "924791712944099389"
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "logo"
+    },
+    { pathname } = document.location,
+    paths = pathname.split("/");
+  if (!paths[0]) paths.shift();
+
+  if (pathname === "/") presenceData.details = "Viewing the home page";
+  else if (pathname.startsWith("/messages")) {
+    if (!paths[1]) presenceData.details = "Viewing messages";
+    else if (paths[1].startsWith("pm-")) {
+      const body = document.querySelector(
+          "body > main.animated > div.wrapper > div.dialogPadding"
+        ),
+        username = body.querySelector("span.username").textContent;
+
+      presenceData.smallImageKey = parseAvatarFromAttr(
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Messaging ${username}`;
+      presenceData.smallImageText = username;
+    }
+  } else if (pathname === "/chat")
+    presenceData.details = "Chatting in the chat";
+  else if (pathname.startsWith("/otakus")) {
+    if (!paths[1] || paths[1].startsWith("page-")) {
+      presenceData.details = "Viewing otakus";
+      if (paths[1])
+        presenceData.state = `Page ${paths[1].replace("page-", "")}`;
+    } else if (paths[1]) {
+      const body = document.querySelector(
+          "body > main.animated > section.wrapper > div.header"
+        ),
+        username = body.querySelector(
+          "div.content > div.holder > span.heading"
+        ).textContent;
+
+      presenceData.largeImageKey = parseAvatarFromAttr(
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Viewing otaku ${username}`;
+      presenceData.smallImageKey = "logo";
+      presenceData.smallImageText = username;
+    }
+  } else if (pathname.startsWith("/animes")) {
+    if (paths[1] === "search" && paths[2])
+      presenceData.details = `Searching for ${paths[2].replaceAll("-", " ")}`;
```

Comment:
```suggestion
      presenceData.details = "Searching for";
      presenceData.state = paths[2].replaceAll("-", " ")
```

Reply:
showing that only in `details` is looking better, already tried

Reply:
It won't fit on anybody's client lmao

Reply:
ok, lets when is more than 10 characters to be in the `state` and when is less than 10 to be in `details`, ok?

Reply:
![image](https://user-images.githubusercontent.com/62988428/147686991-26afe0f6-6c21-402e-8ad5-35cca903032e.png)
![image](https://user-images.githubusercontent.com/62988428/147687012-864e47ed-6e0e-44ce-954f-64f72017d195.png)